app/src/main/java/com/jetbrains/handson/mpp/mobile/AdvancedSearchActivity.kt

```diff
@@ -29,61 +31,76 @@ class AdvancedSearchActivity : AppCompatActivity(), DatePickerDialog.OnDateSetLi
         var myMinute: Int = 0
         var timeChanged = false
 
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_pop_up_window)
 
-        val exitPopUpButton = findViewById<Button>(R.id.submit_advanced_search)
-        exitPopUpButton.setOnClickListener {
-            val intent = Intent(this, MainActivity::class.java)
-            intent.putExtra("numAdults", getNumAdults() )
-            intent.putExtra("numChildren", getNumChildren())
-            if (timeChanged){
-                intent.putExtra("time", "$myYear-${makeNumberTwoDigits(myMonth)}-${makeNumberTwoDigits(myDay)}T${makeNumberTwoDigits(myHour)}:${makeNumberTwoDigits(myMinute)}:00.000")
-            } else {
-                intent.putExtra("time", "Unset")
-            }
-            setResult(2, intent)
-            finish()
-        }
+        presenter = AdvancedSearchPresenter()
+        presenter.onViewTaken(this)
 
-        textView = findViewById(R.id.in_date)
-        button = findViewById(R.id.btn_date)
-        button.setOnClickListener {
-            val calendar: Calendar = Calendar.getInstance()
-            day = calendar.get(Calendar.DAY_OF_MONTH)
-            month = calendar.get(Calendar.MONTH)
-            year = calendar.get(Calendar.YEAR)
-            val datePickerDialog =
-                DatePickerDialog(this@AdvancedSearchActivity, this@AdvancedSearchActivity, year, month,day)
-            datePickerDialog.show()
-        }
+        setInitialValues()
+        setListeners()
+
+    }
 
+    private fun setInitialValues() {
         val numAdultsView = findViewById<TextView>(R.id.in_num_adults)
         numAdultsView.text = "1"
         val numChildrenView = findViewById<TextView>(R.id.in_num_children)
         numChildrenView.text = "0"
         val timeView = findViewById<TextView>(R.id.in_date)
         timeView.text = "Today, Now"
+    }
 
+    private fun setListeners() {
+        val exitPopUpButton = findViewById<Button>(R.id.submit_advanced_search)
+        exitPopUpButton.setOnClickListener {
+            submitAdvancedSearch()
+        }
+
+        textView = findViewById(R.id.in_date)
+        val dateButton = findViewById<Button>(R.id.btn_date)
+        dateButton.setOnClickListener {
+            showDatePicker()
+        }
+
+        val numAdultsView = findViewById<TextView>(R.id.in_num_adults)
+        val numChildrenView = findViewById<TextView>(R.id.in_num_children)
         add_num_adults.setOnClickListener {
```

Comment:
Are there any limits you might want to set on the allowed values of adults / children? Is there a maximum number of passengers allowed in a search? Is there a minimum?